usr/lib/soft_stdll/soft_specific.c

```diff
@@ -2751,18 +2751,15 @@ static CK_RV softtok_hmac_final(SIGN_VERIFY_CONTEXT *ctx, CK_BYTE *signature,
         return CKR_MECHANISM_INVALID;
     }
 
-
-if (signature == NULL) 
-	{	
-    	if (sign) {
-      	  if (general)
-            *sig_len = *(CK_ULONG *) ctx->mech.pParameter;
-     	  else
-            *sig_len = (CK_ULONG)mac_len;
-	
-	 return CKR_OK;
-	}
+    if (signature == NULL) {
+        if (sign) {
+      	    if (general)
```

Comment:
Better, but still some tabs...
In line 'if (general)' is a tab, and in line '*sig_len = (CK_ULONG)mac_len;' is a tab at the end of the line. Also in line 'return CKR_OK' and the following line with just a closing bracket.  There is also a tab in the empty line after the code block you added. I would suggest to use an editor that shows whitespace characters. That way you can quickly see the tabs....